File: Day236_2025/Q236_Leetcode.java
class Q236_Leetcode {
    public int longestSubarray(int[] nums) {
        int zeroCount = 0;
        int longestWindow = 0;
        int start = 0;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) zeroCount++;
            
            while (zeroCount > 1) {
                if (nums[start] == 0) zeroCount--;
                start++;
            }
            
            longestWindow = Math.max(longestWindow, i - start);
        }
        
        return longestWindow;
    }
}

File: Day236_2025/Q236_Leetcode.py
from typing import List

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        zero_count = 0
        longest_window = 0
        start = 0
        
        for i, num in enumerate(nums):
            if num == 0:
                zero_count += 1
                
            while zero_count > 1:
                if nums[start] == 0:
                    zero_count -= 1
                start += 1
                
            longest_window = max(longest_window, i - start)
        
        return longest_window
